downloader/src/main/kotlin/vcs/GitWorkingTree.kt

```diff
@@ -53,11 +53,12 @@ private fun findGitOrSubmoduleDir(workingDirOrFile: File): Repository {
 open class GitWorkingTree(workingDir: File, vcsType: VcsType) : WorkingTree(workingDir, vcsType) {
     companion object : Logging
 
-    private val repo = findGitOrSubmoduleDir(workingDir)
+    private val repo: Repository
```

Comment:
IIUC the constructor threw an exception previously, and as of now not anymore.
Shall we ensure it fails early again?

Reply:
I actually think it's even better the new way: There might be cases where you'd like to create a `GitWorkingTree` instance that points to a `workingDir` that yet needs to be initialized (maybe even in the scope of file existence checks in rules).

Reply:
Hm, I believe failing early is good by default unless there is a particular case which satisfies not doing so.
I'd prefer keeping fail early until we come across a case which requires it's removal. But am not going to block this.

Reply:
> I believe failing early is good by default

I agree, but what I'm trying to say is that creating a `GitWorkingTree` instance wrapping a directory that is no Git tree yet is not necessarily a failure. It only has to be a Git tree once a Git operation is executed on it.

That's BTW also how the other (purely CLI-based) `WorkingTree` implementations work: They will only throw an exception at the time a CLI tool is run and fails, not at instance construction time. So this commits also aligns the behavior across `WorkingTree` implementations.